packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,176 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'text_field.dart';
+import 'icons.dart';
+
+// This is a light gray color used for UISearchTextField's background.
+const Color _kDefaultBackgroundColor = Color.fromRGBO(239, 239, 240, 1.0);
+
+// This is approximately the corner radius used in UISearchTextField.
+// Note that UISearchTextField uses continuous corners, as opposed to circular.
+const BorderRadius _kDefaultBorderRadius =
+    BorderRadius.all(Radius.circular(9.0));
+
+// This is a dark gray color used for UISearchTextField's tint color.
+const Color _kDefaultItemColor = Color.fromRGBO(132, 132, 136, 1.0);
+
+// This is a dark gray color used for UISearchTextField's placeholder color.
+const TextStyle _kDefaultPlaceholderStyle =
+    TextStyle(color: Color.fromRGBO(142, 142, 147, 1.0));
+
+// These are insets used to mimick UISearchTextField's search icon placement.
+const EdgeInsets _kDefaultPrefixInsets = EdgeInsets.fromLTRB(6, 0, 0, 4);
+
+// These are insets used to mimick UISearchTextField's X-Mark icon placement.
+const EdgeInsets _kDefaultSuffixInsets = EdgeInsets.fromLTRB(0, 0, 5, 2);
+
+// These are insets used to mimick UISearchTextField's text placement.
+const EdgeInsets _kDefaultContentPadding = EdgeInsets.fromLTRB(3.8, 8, 5, 8);
+
+// The suffix mode determines whether the X-Mark appears during editing.
+const OverlayVisibilityMode _kDefaultSuffixMode = OverlayVisibilityMode.editing;
+
+// This is a decoration that uses the default background color and border radius
+const BoxDecoration _kDefaultBoxDecoration = BoxDecoration(
+  color: _kDefaultBackgroundColor,
+  borderRadius: _kDefaultBorderRadius,
+);
+
+// The default placeholder seen in UISearchTextField.
+const String _kDefaultPlaceholder = 'Search';
+
+// This returns the icon sizes, which scales with the accessibility font size.
+double _scaledIconSize(BuildContext context) {
+  return MediaQuery.textScaleFactorOf(context) * 20.0;
+}
+
+// Creates the search icon which is used as a prefix.
+Widget _defaultPrefix(Color itemColor) {
+  return Builder(
+    builder: (BuildContext context) => Icon(
+      CupertinoIcons.search,
+      color: itemColor,
+      size: _scaledIconSize(context),
+    ),
+  );
+}
+
+// Creates the X-Mark icon which is used as a suffix. It accepts a tap action.
+Widget _defaultSuffix(Color itemColor, Function() onTap) {
+  return Builder(
+    builder: (BuildContext context) => GestureDetector(
```

Comment:
Left this comment up even though it's outdated. I've put a TODO in the updated code for this. It was really tricky for me to get the button to have truly zero padding (so that it doesn't effect the search field's width.)